adapter/PreciceInterface.c

```diff
@@ -670,44 +682,47 @@ void PreciceInterface_ConfigureCouplingData(PreciceInterface *interface, Simulat
+      interface->fluxDataID = precicec_getDataID("Heat-Flux", interface->faceCentersMeshID);
       printf("Write data '%s' found with ID # '%d'.\n", config->writeDataNames[i], interface->fluxDataID);
     } else if (startsWith(config->writeDataNames[i], "Sink-Temperature-")) {
-      interface->writeData[i]                 = SINK_TEMPERATURE;
+      interface->writeData[i] = SINK_TEMPERATURE;
+      PreciceInterface_EnsureValidFacesMeshID(interface, "Sink temperature");
       interface->kDeltaTemperatureWriteDataID = precicec_getDataID(config->writeDataNames[i], interface->faceCentersMeshID);
       printf("Write data '%s' found with ID # '%d'.\n", config->writeDataNames[i], interface->kDeltaTemperatureWriteDataID);
     } else if (startsWith(config->writeDataNames[i], "Heat-Transfer-Coefficient-")) {
-      interface->writeData[i]      = HEAT_TRANSFER_COEFF;
+      interface->writeData[i] = HEAT_TRANSFER_COEFF;
+      PreciceInterface_EnsureValidFacesMeshID(interface, "Heat Transfer Coefficient");
       interface->kDeltaWriteDataID = precicec_getDataID(config->writeDataNames[i], interface->faceCentersMeshID);
       printf("Write data '%s' found with ID # '%d'.\n", config->writeDataNames[i], interface->kDeltaWriteDataID);
     } else if (startsWith(config->writeDataNames[i], "DisplacementDelta")) {
-      PreciceInterface_EnsureValidNodesMeshID(interface);
+      PreciceInterface_EnsureValidNodesMeshID(interface, "DisplacementDeltas");
       interface->writeData[i]             = DISPLACEMENTDELTAS;
       interface->displacementDeltasDataID = precicec_getDataID(config->writeDataNames[i], interface->nodesMeshID);
       printf("Write data '%s' found with ID # '%d'.\n", config->writeDataNames[i], interface->displacementDeltasDataID);
     } else if (startsWith(config->writeDataNames[i], "Displacement")) {
-      PreciceInterface_EnsureValidNodesMeshID(interface);
+      PreciceInterface_EnsureValidNodesMeshID(interface, "Displacement");
       interface->writeData[i]        = DISPLACEMENTS;
       interface->displacementsDataID = precicec_getDataID(config->writeDataNames[i], interface->nodesMeshID);
       printf("Write data '%s' found with ID # '%d'.\n", config->writeDataNames[i], interface->displacementsDataID);
     } else if (startsWith(config->writeDataNames[i], "Position")) {
-      PreciceInterface_EnsureValidNodesMeshID(interface);
+      PreciceInterface_EnsureValidNodesMeshID(interface, "Position");
       interface->writeData[i]    = POSITIONS;
       interface->positionsDataID = precicec_getDataID(config->writeDataNames[i], interface->nodesMeshID);
       printf("Write data '%s' found with ID # '%d'.\n", config->writeDataNames[i], interface->positionsDataID);
-    } else if (startsWith(config->writeDataNames[i], "Velocit")) {
-      PreciceInterface_EnsureValidNodesMeshID(interface);
+    } else if (startsWith(config->writeDataNames[i], "Velocity")) {
+      PreciceInterface_EnsureValidNodesMeshID(interface, "Velocity");
```

Comment:
I think this `Velocit` was actually the intended fix, to also work with `Velocities`. Since that already confused you, a comment would be useful here.

Reply:
Oooh right, I misunderstood the thing. For the record, explanation is here: https://github.com/precice/calculix-adapter/issues/64
I'm undoing this change.